src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,177 @@
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this , this.findListView);
+    }
+
+    public static translateMessageStatus(fromValue: number, toValue: number, maxPageSize: number): string {
+        return window.mx.ui.translate("mxui.lib.MxDataSource", "status", [ fromValue, toValue, maxPageSize ]);
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-pagination", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            createElement(Alert, {
+                className: "widget-pagination-alert",
+                message: this.state.message
+            }),
+            this.renderPageButton()
+        );
+    }
+
+    componentDidMount() {
+        const queryNode = findDOMNode(this) as HTMLElement;
+        const targetNode = findTargetNode(queryNode);
+        this.hideLoadMoreButton(targetNode);
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    private renderPageButton(): ReactElement<PaginationProps> | null {
+        if (this.state.validationPassed) {
+            return createElement(Pagination, {
+                getMessageStatus: PaginationContainer.translateMessageStatus,
+                hideUnusedPaging: this.state.hideUnusedPaging,
+                items: this.props.items,
+                listViewSize: this.state.listViewSize,
+                maxPageButtons: this.props.maxPageButtons,
+                offset: this.state.offset,
+                onClickAction: this.updateListView,
+                pagingStyle: this.props.pagingStyle
+            });
+        }
+
+        return null;
+    }
+
+    private findListView() {
+        if (this.state.findingListViewWidget) {
+            const queryNode = findDOMNode(this) as HTMLElement;
+            const targetNode = findTargetNode(queryNode);
+            let hideUnusedPaging = false;
+            let targetListView: ListView | null = null;
+            let listViewSize = 0;
+            let offset = 0;
+            let dataSource: ListView["_datasource"];
+
+            if (targetNode) {
+                this.hideLoadMoreButton(targetNode);
+                targetListView = dijitRegistry.byNode(targetNode);
+
+                if (targetListView) {
+                    dataSource = targetListView._datasource;
+                    listViewSize = dataSource._setSize;
+                    offset = targetListView._datasource._pageSize;
+                    hideUnusedPaging = (offset >= dataSource._setSize) && this.props.hideUnusedPaging;
+
+                    dojoAspect.after(targetListView, "_onLoad", () => {
+                        if (this.state.targetListView) {
+                            this.setState({
+                                listViewSize: this.state.targetListView._datasource._setSize,
+                                offset
```

Comment:
How does this widget support when the amount of pages are changes? Add new object delete new object?
Please add a test case in the project for this.

Reply:
It will work given that already on the search its working due to this; Searching from the list view search changes listview structure
```
dojoAspect.after(targetListView, "_onLoad", () => {
                        if (this.state.targetListView) {
                            this.setState({
                                listViewSize: this.state.targetListView._datasource._setSize,
                                offset
                            });
                        }
                    });

```

Reply:
However i have added a test case